x-pack/platform/plugins/shared/ml/public/application/data_frame_analytics/pages/analytics_creation/hooks/use_index_data.ts

```diff
@@ -81,54 +82,40 @@ export const useIndexData = (
   toastNotifications: CoreSetup['notifications']['toasts'],
   runtimeMappings?: RuntimeMappings
 ): UseIndexDataReturnType => {
+  const isMounted = useMountedState();
+  const {
+    services: {
+      data: { dataViews: dataViewsService },
+    },
+  } = useMlKibana();
   const mlApi = useMlApi();
-  // Fetch 500 random documents to determine populated fields.
-  // This is a workaround to avoid passing potentially thousands of unpopulated fields
-  // (for example, as part of filebeat/metricbeat/ECS based indices)
-  // to the data grid component which would significantly slow down the page.
   const [dataViewFields, setDataViewFields] = useState<string[]>();
   const [timeRangeMs, setTimeRangeMs] = useState<TimeRangeMs | undefined>();
 
   useEffect(() => {
-    async function fetchDataGridSampleDocuments() {
+    async function fetchPopulatedFields() {
       setErrorMessage('');
       setStatus(INDEX_STATUS.LOADING);
 
-      const esSearchRequest = {
-        index: dataView.title,
-        body: {
-          fields: ['*'],
-          _source: false,
-          query: {
-            function_score: {
-              query: { match_all: {} },
-              random_score: {},
-            },
-          },
-          size: 500,
-        },
-      };
-
       try {
-        const resp: IndexSearchResponse = await mlApi.esSearch(esSearchRequest);
-        const docs = resp.hits.hits.map((d) => getProcessedFields(d.fields ?? {}));
+        const indexPattern = dataView.getIndexPattern();
+        const nonEmptyFields = await dataViewsService.getFieldsForWildcard({
+          pattern: indexPattern,
+          includeEmptyFields: false,
+        });
 
-        // Get all field names for each returned doc and flatten it
-        // to a list of unique field names used across all docs.
-        const allDataViewFields = getFieldsFromKibanaDataView(dataView);
-        const populatedFields = [...new Set(docs.map(Object.keys).flat(1))]
-          .filter((d) => allDataViewFields.includes(d))
-          .sort();
+        const populatedFields = nonEmptyFields.map((field) => field.name);
 
-        setStatus(INDEX_STATUS.LOADED);
-        setDataViewFields(populatedFields);
+        if (isMounted()) {
+          setDataViewFields(getPopulatedFieldsFromKibanaDataView(dataView, populatedFields));
```

Comment:
On `main` we only show the populated fields in the DFA data grid:

<img width="514" alt="Screenshot 2025-01-08 at 13 13 26" src="https://github.com/user-attachments/assets/d6e5c1eb-aba5-4a3e-8de6-6459d113bcbf" />

But here we are showing all fields

<img width="535" alt="Screenshot 2025-01-08 at 13 04 59" src="https://github.com/user-attachments/assets/922945b6-6285-4eaf-a284-d53f18432cf1" />

Transforms seems to be correctly only showing the populated fields.

Reply:
This is fixed in the data grid by https://github.com/elastic/kibana/pull/205417/commits/1f03150274a6425b27f672c7ac226c5deef1f65d, but populated runtime fields are showing up as being non-populated in the wizard form, such as the dependent variable control for regression or classification jobs.

Reply:
Confirmed that populated runtime fields are now showing up correctly in the DFA forms.